### PR TITLE
added save manager and super basic save data class

### PR DESCRIPTION
**Pull request by nshooter3**
Added savemanager for initializing, saving, and loading save files. The save file itself is handled via a serializable save data class, allowing us to track new parameters simply by adding them to the file. Due to this setup, we can reference save params by name through intellisense when changing them. When the game is a saved or loaded, the current instance of the SaveData class and its parameters are converted to or from binary.

Possible issues include support for old save files if we patch the game to add new parameters, though this may be a problem better addressed once the game is further along.

### CODE REVIEW

**Review by mitjmcc — CHANGES_REQUESTED** [review]
Looks good and simple.
Two things though:
- Might as well add a game version int/string. It'll help with updates and is a simple thing to include now to get us started.
- Some error handling around the file I/O? Some try/catch statements would do. We can decide what to actually do in the catch statements later

**Review by mitjmcc — APPROVED** [review]

File: Assets/Scripts/Saving/SaveDataManager.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                using (file = File.Create(Application.persistentDataPath + "/saveData" + fileNum + ".dat"))
+                {
+                    bf = new BinaryFormatter();
+                    bf.Serialize(file, saveData);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error saving file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
```
Comment (Matthew-DeLong) [inline Assets/Scripts/Saving/SaveDataManager.cs]: We should actually have some kind of error handling here. Maybe the return type to a bool and leave it to whatever is calling this function?

File: Assets/Scripts/Saving/SaveDataManager.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    bf = new BinaryFormatter();
+                    SaveData initSaveData = new SaveData();
+                    initSaveData.currentSaveFileActive = fileNum;
+                    bf.Serialize(file, initSaveData);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error initializing file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
```
Comment (Matthew-DeLong) [inline Assets/Scripts/Saving/SaveDataManager.cs]: We should actually have some kind of error handling here. Maybe the return type to a bool and leave it to whatever is calling this function?

File: Assets/Scripts/Saving/SaveDataManager.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                using (file = File.Open(Application.persistentDataPath + "/saveData" + fileNum + ".dat", FileMode.Open))
+                {
+                    bf = new BinaryFormatter();
+                    saveData = (SaveData)bf.Deserialize(file);
+                    file.Close();
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Error loading file " + fileNum + "\n" + e.Message + "\n" + e.StackTrace);
```
Comment (Matthew-DeLong) [inline Assets/Scripts/Saving/SaveDataManager.cs]: We should actually have some kind of error handling here. Maybe the return type to a bool and leave it to whatever is calling this function?

**Review by Matthew-DeLong — APPROVED** [review]
My three comments are actually one comment, but I'm happy with this as-is. It looks very easy to add to the SaveData, so I'm not worried about trying get everything we need in it right away.

**Comment by nshooter3** [comment]
SaveDataManager now returns a bool and has an out parameter for the error message instead of throwing an exception.

**Review by Matthew-DeLong — APPROVED** [review]

**Pull request merged**